### PR TITLE
Make ossl_time_now() a static inline function, and move ossl_sleep()

### PR DESCRIPTION
**Pull request by levitte**
To make the whole `OSSL_TIME` API usable from anywhere with no dependencies
other than publicly exposed types and functions, `ossl_time_now()` is made a
static inline function alongside the other `ossl_time` functions.

Furthermore, `ossl_sleep()` is moved to its own header, to avoid interlocking
dependencies between `include/internal/time.h` and `include/internal/e_os.h`.

### CODE REVIEW

**Comment by levitte** [comment]
This is an alternative to #19330, and perhaps less controversial since it
doesn't expose any new functions to the public[^1]

[^1]: Not that I really understand why that's an issue in the master branch...

**Comment by levitte** [comment]
Since #19330 is up for an OTC discussion, this PR should be part of the same discussion.  Labeling it accordingly.

File: test/event_queue_test.c [hunk 1]
```diff
@@ -7,17 +7,20 @@
  * https://www.openssl.org/source/license.html
  */
 
+/* Custom implementation of ossl_time_now() */
+#define IMPLEMENT_OSSL_TIME_NOW                 \
```
Comment (paulidale) [inline test/event_queue_test.c:11]: Great idea!

**Review by paulidale — APPROVED** [review]

**Comment by t-j-h** [comment]
I really don't like this approach. The concept of using inline functions for this sort of thing (portability) I dislike on principle. 
Exposing the function makes sense IMHO as that is a simple solution.
Digging the use-internal-headers-in-apps hole deeper is also not really the attractive path IMHO.

**Comment by beldmit** [comment]
+1 to @t-j-h 
I don't see any problems with any new API function.

**Comment by levitte** [comment]
I agree, @t-j-h.  I'm only offering this PR as an alternative if the OTC decides that #19330 is a no go

**Comment by levitte** [comment]
#19330 was approved by the OTC, so dropping this one in favor of that

**Pull request closed without merging**